menubot/index.js

```diff
@@ -1,85 +1,102 @@
-import {BossRoomBot, CharInfo} from "../bossroom";
-import {RGValidator, RGBot} from "../rg";
+import { CharInfo } from "../bossroom";
 
-let charType = 1; // fixed to rogue character
+let rg = null;
 
-export function configureBot(characterType) {
-  console.log(`Unity bot configureBot function called, charType: ${charType} - characterType: ${characterType}`);
-  charType = CharInfo.type.indexOf(characterType);
+/**
+ * Defines the type of character that the game should use for this bot.
+ */
+ export function getCharacterType() {
+  return CharInfo.type[1]; // fixed to rogue character
 }
 
-let rgValidator = new RGValidator();
+/**
+ * One of ...
+ * MANAGED - Server disconnects/ends bot on match/game-scene teardown
+ * PERSISTENT - Bot is responsible for disconnecting / ending itself
+ */
+export function getBotLifecycle() {
+  return 'PERSISTENT';
+}
+
+/**
+ * @returns {boolean} true if I'm an in-game character, or false if I'm an invisible navigator/observer/etc.
+ */
+export function isSpawnable() {
+  return false;
+}
+
+/**
+ * Let server know if i have finished my processing
+ * @returns {boolean} true if done processing and ready to be torn down
+ */
+export function isComplete() {
+  return rg ? rg.isComplete() : false;
+}
 
-let botComplete = false;
 
 // flags for clicking the 6 buttons we need to click to start the game
 let stateFlags = {
-  "RGHostButton":false,
-  "StartWithRGButton":false,
-  "SelectProfileButton":false,
-  "ProfileMenuButton":false,
-  "ReadyButton":false,
-  "Seat7Button":false,
+  "RGHostButton": false,
+  "StartWithRGButton": false,
+  "SelectProfileButton": false,
+  "ProfileMenuButton": false,
+  "ReadyButton": false,
+  "Seat7Button": false,
 }
 
 let playedGame = false;
 
-export async function runTurn(playerId, tickInfo, mostRecentMatchInfo, actionQueue) {
-
-  // On each state, run through the validations and see if any failed or passed
-  rgValidator.checkValidations(tickInfo);
-
-  const t = tickInfo.tick;
-  const sceneName = tickInfo.sceneName;
+export async function configureBot(rgObject) {
+  rg = rgObject;
```

Comment:
i feel like this should be a no-op ?

why wouldn't we use the rgObject reference passed into runTurn

Reply:
Good call, updated